feature/agenda/presentation/src/main/java/com/sergius/agenda/presentation/task/TaskDetailsViewModel.kt

```diff
@@ -1,6 +1,35 @@
 package com.sergius.agenda.presentation.task
 
+import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.stateIn
 
+@OptIn(ExperimentalMaterial3Api::class)
 class TaskDetailsViewModel: ViewModel() {
+
+    private val _state = MutableStateFlow(TaskDetailsState())
+    val state = _state
+        .onStart {  }
+        .stateIn(
+            scope = viewModelScope,
+            started = kotlinx.coroutines.flow.SharingStarted.WhileSubscribed(2000),
+            initialValue = TaskDetailsState()
+        )
+
+    fun onAction(action: TaskDetailsAction) {
+        when(action) {
+            is TaskDetailsAction.OnToggleTimerDialogVisibility -> {
+                _state.value = _state.value.copy(showTimerDialog = !_state.value.showTimerDialog)
+            }
+
+            is TaskDetailsAction.OnToggleDateDialogVisibility -> {
+                _state.value = _state.value.copy(showDateDialog = !_state.value.showDateDialog)
+            }
```

Comment:
I think I've commented this before, but in general I'd recommend using `_state.update` instead. It's just as simple and avoids race conditions (you don't have any multithreading here, but using `update` doesn't have any downsides, so I'd just generally stick to that)